app/modules/sightings/models.py

```diff
@@ -94,6 +94,8 @@ def delete_from_edm_by_guid(cls, current_app, guid):
     # given edm_json (verbose json from edm) will populate with houston-specific data from feather object
     # note: this modifies the passed in edm_json, so not sure how legit that is?
     def augment_edm_json(self, edm_json):
+        edm_json['createdHouston'] = self.created.isoformat()
+        edm_json['updatedHouston'] = self.updated.isoformat()
```

Comment:
This looks fine. 
The bit that does concern me is the  
self.encounters[i].augment_edm_json(edm_json['encounters'][i])
line below as it implicitly assumes that the order that the encounters are received from EDM is the same as in houston. Are we utterly certain that this is a valid assumption?

Reply:
if the order is different, something is wrong (the same way if the _count_ is different, something is wrong).  thats not to say it wont happen.  but the sighting in *edm* should be made with the exact same json that creates them in *houston*, thus it should be _created_ in the same order.  there is (of course) always potential for them to then diverge, which is an ongoing concern in a lot of places for sure.

making it more robust here might be a good thing .. or maybe we would want it to be more robust in reporting things are _out of sync_?  e.g. what if we found an encounter was missing etc.   i have no doubt we are going to find weirdness eventually with all things that sync between 2+ systems in codex.   what i am unsure of is the best way to handle when they are!

perhaps a compromise for now would be robustly handle the case where _the exact right number of exactly the same encounters_ are there, but in the wrong order.  but then throw exceptions for every other broken case?

Reply:
i went ahead and allow for arbitrary ordering on encounters now; throws exceptions if something is off now (rather than gracefully skipping things)  :+1: